### PR TITLE
update-dns configuration file

### PR DESCRIPTION
**Pull request by Morley93**
Reading update-dns settings from a JSON configuration file

    * External IP URL, personal access token, (sub)domain and poll frequency read from file
    * Root config file at /etc/dns-dodo.conf, user at ~/.dns-dodo.conf
    * Flags override any settings derived from a config file
    * systemd unit file allows running as a service

### CODE REVIEW

File: dns-dodo.service [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=The dns-dodo DNS record updater
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=/usr/bin/dns-dodo update-dns -p
```
Comment (lummie) [inline dns-dodo.service:7]: lol, noticed you changed from your home dir. Current readme instructions say how to build which ends up in $GOPATH/bin Maybe add a comment to this file to update the path or copy $GOPATH/bin/dns-dodo to /usr/bin?

Comment (Morley93) [reply dns-dodo.service:7]: Ahh damn, I thought I'd gotten away with it! 😉

The reason I put `/usr/bin` is that once the unit file is in I'd like to provide a `PKGBUILD` file for Arch Linux AUR which would install to `/usr/bin`. However, that would mean having to package for each distro individually. I will investigate using systemd's user units in combination with `$GOPATH` for a cross-distro solution.

Comment (Morley93) [reply dns-dodo.service:7]: After some research, I don't think providing a systemd unit file that is compatible with `go install` is all that practical. Part of reason is that while you can _specify_ environment variables within systemd unit files, you can't access existing variables or use variables to construct relative paths.

We could get around this by including a launch script which uses an environment variable set in the unit (`GOPATH` in this case), but this still doesn't solve much - the user would still have to manually set their `GOPATH` in the unit and copy/link it to `~/.local/share/systemd/user`.

We could think about providing a unit template (or putting one in the README), but I think the existing unit file pointing at `/usr/bin` for an AUR build should be included nonetheless.

Comment (Morley93) [reply dns-dodo.service:7]: Added a template unit file and a script to install it as a user unit. This should be sufficient to start the binary installed with `go install` as a systemd service.

**Comment by lummie** [comment]
Great addition :)

**Pull request merged**